### DISCUSSION TITLE
Problem with struct.error: unpack requires a buffer of 4 bytes

### DISCUSSION THREAD

**Discussion by AntonWert** [pymodbus-dev/pymodbus#1614] [Help] [answered]
Hello,
first of all, I am not an expert in python at all, so maybe this is simple for you - but I am currently stuck and would appreciate if I can get help.

My setup: RPi3 with 32-bit raspbian OS running PyModMon (https://github.com/transistorgrab/PyModMon) to pull data from an SMC solar inverter. PyModMon  is using pymodbus.
Everything works fine.

Now I want to move to RPi4 with 64-bit OS and just running the same setup.

But I do get this error, just when calling exactly the same command:

    interpreted = message.decode_32bit_uint()
  File "/usr/local/lib/python3.9/dist-packages/pymodbus/payload.py", line 382, in decode_32bit_uint
    handle = self._unpack_words(fstring, handle)
  File "/usr/local/lib/python3.9/dist-packages/pymodbus/payload.py", line 342, in _unpack_words
    handle = unpack(f"!{wc_value}H", handle)
struct.error: unpack requires a buffer of 4 bytes

Is this a bug in pymodbus, or what do I do wrong?

**Comment by janiversen** [comment, marked as answer]
What are you trying to do ?

that method is not meant for external use. But the problem is clear, the message contains only 16bit not 32, probably because you only read 1 register (32 bit is 2 registers).

**Comment by janiversen** [comment]
Apart from that if you are using pyModMon and not the newer pyModMon_3 you will probably encounter a number of problems.

I looked into the pyModMon_3 code, and the problem you experience is in their code, not our code, so please ask them to solve it.

**Reply by AntonWert** [reply]
Ok, thank you for your proposal
